File: 549/549.py
from typing import List


class Solution:
    def findLHS(self, nums: List[int]) -> int:
        nums_set = {}
        output = 0
        for num in nums:
            if num not in nums_set:
                nums_set[num] = 0
            nums_set[num] += 1
        for k, v in nums_set.items():
            if k+1 in nums_set:
                output = max(output, nums_set[k] + nums_set[k+1])
        return output

File: leetcode/549/549.go
package main

func findLHS(nums []int) int {
	output := 0
	nums_set := make(map[int]int)
	for _, num := range nums {
		if _, exist := nums_set[num]; !exist {
			nums_set[num] = 0
		}
		nums_set[num]++
	}

	for k, count := range nums_set {
		if count2, exist := nums_set[k+1]; exist {
			output = max(output, count+count2)
		}
	}
	return output
}
